src/x/sync/pooled_worker_pool_test.go

```diff
@@ -50,6 +51,35 @@ func TestPooledWorkerPoolGo(t *testing.T) {
 	require.Equal(t, uint32(testWorkerPoolSize*2), count)
 }
 
+func TestPooledWorkerPoolGoWithContext(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	wp, err := NewPooledWorkerPool(testWorkerPoolSize,
+		NewPooledWorkerPoolOptions().SetGrowOnDemand(false))
+	require.NoError(t, err)
+	wp.Init()
+
+	// Cancel and make sure worker will prefer to return from cancelled
+	// work rather than always enqueue.
+	cancel()
+
+	var aborted uint32
+	for i := 0; i < 100; i++ {
+		go func() {
+			result := wp.GoWithContext(ctx, func() {
+				time.Sleep(time.Second)
```

Comment:
should we make these block instead of sleep so that we can assert on a specific # of workers that cancel? Alternatively if we check cancel first, we can assert that abort == 100, I think...

Reply:
Likely, although unfortunately the pooled worker has a lot of quirks that makes it hard to know actually how many workers will be spun up (e.g. the number of shards and `now.UnixNano()` by number of shards).

This should be sufficient for the time being.